server/src/main/java/org/apache/iotdb/db/mpp/sql/analyze/DataNodeSchemaCache.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.mpp.sql.analyze;
+
+import org.apache.iotdb.commons.utils.TestOnly;
+import org.apache.iotdb.db.conf.IoTDBConfig;
+import org.apache.iotdb.db.conf.IoTDBDescriptor;
+import org.apache.iotdb.db.metadata.path.MeasurementPath;
+import org.apache.iotdb.db.mpp.common.schematree.SchemaNode;
+import org.apache.iotdb.db.mpp.common.schematree.SchemaTree;
+
+import com.github.benmanes.caffeine.cache.CacheLoader;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.github.benmanes.caffeine.cache.RemovalListener;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class takes the responsibility of metadata cache management of all DataRegions under
+ * StorageEngine
+ */
+public class DataNodeSchemaCache {
+  private static final Logger logger = LoggerFactory.getLogger(DataNodeSchemaCache.class);
+
+  private static IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+  private SchemaTree root;
+
+  private LoadingCache<MeasurementPath, SchemaNode> schemaNodeCache;
+
```

Comment:
It seems all the key of this cache already maintains all the schema info we need. We implement this schemaTree to compress the prefix of series and this feature is broken by the schemaNodeCache.

Reply:
use partialPath ->SchemaEntity after out talk.